gapii/client/capture.go

```diff
@@ -146,6 +153,79 @@ func (p *Process) connect(ctx context.Context, gvrHandle uint64, interceptorPath
 	})
 }
 
+func handleCommError(ctx context.Context, commErr error, anyDataReceived bool) (abort bool, err error) {
+	switch {
+	case errors.Cause(commErr) == io.EOF:
+		log.E(ctx, "unexpected end of stream")
+		abort = true
+		err = commErr
+	case commErr != nil && anyDataReceived:
+		netErr, isnet := commErr.(net.Error)
+		if !isnet || (!netErr.Temporary() && !netErr.Timeout()) {
+			log.E(ctx, "Connection error: %v", commErr)
+			// Got an error mid-stream terminate.
+			abort = true
+			err = commErr
+		}
+	case commErr != nil && !anyDataReceived:
+		log.E(ctx, "Target not ready: %v", commErr)
+		// Got an error without receiving a byte of data.
+		// Treat failure-to-connect as target-not-ready instead of an error.
+		abort = true
+	}
+	return
+}
+
+func readHeader(conn net.Conn) (msgType messageType, dataSize uint64, err error) {
+	msgType = messageInvalid
+	now := time.Now()
+	conn.SetReadDeadline(now.Add(time.Millisecond * 500)) // Allow for stop event and UI refreshes.
+	var bufSize int
+	buf := make([]byte, 6)
+	bufSize, err = io.ReadFull(conn, buf)
+	if bufSize == 6 {
```

Comment:
More canonical Go would read
```
if bufSize, err = io.ReadFull(conn, buf); err != nil {

}
```

Reply:
Thanks, of course I prefer to be more canonical. But I insist in this case to use 'err == nil' instead :)